### DISCUSSION TITLE
The hierarchy update API throwing error,  "Metadata trackable cannot have new property with name autoBatch"

### DISCUSSION THREAD

**Discussion by vaivk369** [Sunbird-coKreat/Community#83] [Bugs]
In cokreat, when we create project for collections , collections created on ED portals are getting copied to cokreat and later the hierarchy update API is called internally to update the sections of that collection. 
Currently we are facing the issue, where hierarchy update API is giving following error

`{
    "id": "api.collection.hierarchy.update",
    "ver": "4.0",
    "ts": "2023-05-24T09:26:50ZZ",
    "params": {
        "resmsgid": "1afa60df-a7cc-4667-8f29-abae622618d9",
        "msgid": null,
        "err": "CLIENT_ERROR",
        "status": "failed",
        "errmsg": "Validation Errors"
    },
    "responseCode": "CLIENT_ERROR",
    "result": {
        "messages": [
            "Metadata trackable cannot have new property with name autoBatch"
        ]
    }
}`

Here is the CURL for the same

`curl --location --request PATCH 'https://dock.sunbirded.org/api/collection/v1/hierarchy/update' \
--header 'Authorization: Bearer XXX' \
--header 'Content-Type: application/json' \
--data '{
    "request": {
        "data": {
            "nodesModified": {
                "do_11380283102077747214120": {
                    "isNew": true,
                    "root": true,
                    "metadata": {
                        "ownershipType": [
                            "createdBy"
                        ],
                        "copyright": "newDevSetUp2023channel",
                        "year": "2011",
                        "subject": [
                            "Art"
                        ],
                        "channel": "sunbird",
                        "organisation": [
                            "newDevSetUp2023"
                        ],
                        "language": [
                            "English"
                        ],
                        "mimeType": "application/vnd.ekstep.content-collection",
                        "objectType": "Collection",
                        "gradeLevel": [
                            "Class 4"
                        ],
                        "appIcon": "",
                        "primaryCategory": "Digital Textbook",
                        "contentEncoding": "gzip",
                        "generateDIALCodes": "Yes",
                        "contentType": "TextBook",
                        "trackable": {
                            "enabled": "No",
                            "autoBatch": "No"
                        },
                        "audience": [
                            "Student"
                        ],
                        "subjectIds": [
                            "vdn_k-12_subject_art"
                        ],
                        "visibility": "Default",
                        "author": "vvvv",
                        "childNodes": [
                            "do_11380219749893734419",
                            "do_11380219749751193617"
                        ],
                        "discussionForum": {
                            "enabled": "No"
                        },
                        "mediaType": "content",
                        "languageCode": [
                            "en"
                        ],
                        "version": 2,
                        "name": "book2",
                        "mediumIds": [
                            "vdn_k-12_medium_english"
                        ],
                        "code": "org.sunbird.y4HMk5",
                        "credentials": {
                            "enabled": "No"
                        },
                        "description": "Enter description for TextBook",
                        "medium": [
                            "English"
                        ],
                        "copyrightYear": 2022,
                        "dialcodeRequired": "No",
                        "createdFor": [
                            "01379157488516300825"
                        ],
                        "creator": "vvvv",
                        "versionKey": "1684838562014",
                        "framework": "vdn_k-12",
                        "boardIds": [
                            "vdn_k-12_board_pune"
                        ],
                        "createdBy": "842e2f00-16d3-487b-99c2-576667e99f22",
                        "compatibilityLevel": 1,
                        "userConsent": "Yes",
                        "gradeLevelIds": [
                            "vdn_k-12_gradelevel_class4"
                        ],
                        "board": "Pune",
                        "resourceType": "Book",
                        "openForContribution": true,
                        "chapterCountForContribution": 2,
                        "chapterCount": 2,
                        "programId": "5bc03a00-ee67-11ed-a249-d1c2d4936c48",
                        "allowedContentTypes": [
                            "eTextbook",
                            "Explanation Content"
                        ],
                        "origin": "do_11380219689970073616",
                        "originData": {
                            "channel": "01379157488516300825"
                        }
                    }
                },
                "do_11380219749893734419": {
                    "isNew": true,
                    "root": false,
                    "metadata": {
                        "ownershipType": [
                            "createdBy"
                        ],
                        "code": "42bee595-0610-4fb2-9a8d-8500eaa31dba",
                        "credentials": {
                            "enabled": "No"
                        },
                        "channel": "sunbird",
                        "language": [
                            "English"
                        ],
                        "mimeType": "application/vnd.ekstep.content-collection",
                        "objectType": "Content",
                        "primaryCategory": "Textbook Unit",
                        "contentEncoding": "gzip",
                        "generateDIALCodes": "No",
                        "contentType": "TextBookUnit",
                        "dialcodeRequired": "No",
                        "trackable": {
                            "enabled": "No",
                            "autoBatch": "No"
                        },
                        "visibility": "Parent",
                        "discussionForum": {
                            "enabled": "No"
                        },
                        "mediaType": "content",
                        "languageCode": [
                            "en"
                        ],
                        "version": 2,
                        "versionKey": "1684838561882",
                        "compatibilityLevel": 1,
                        "userConsent": "Yes",
                        "name": "Textbook Unit1",
                        "attributions": [],
                        "openForContribution": true,
                        "programId": "5bc03a00-ee67-11ed-a249-d1c2d4936c48",
                        "allowedContentTypes": [
                            "eTextbook",
                            "Explanation Content"
                        ],
                        "origin": "do_11380219749893734419",
                        "originData": {
                            "channel": "01379157488516300825"
                        }
                    }
                },
                "do_11380219749751193617": {
                    "isNew": true,
                    "root": false,
                    "metadata": {
                        "ownershipType": [
                            "createdBy"
                        ],
                        "code": "ee3ffb34-2f69-4d08-b66b-ac78ed0b8750",
                        "credentials": {
                            "enabled": "No"
                        },
                        "channel": "sunbird",
                        "language": [
                            "English"
                        ],
                        "mimeType": "application/vnd.ekstep.content-collection",
                        "objectType": "Content",
                        "primaryCategory": "Textbook Unit",
                        "contentEncoding": "gzip",
                        "generateDIALCodes": "No",
                        "contentType": "TextBookUnit",
                        "dialcodeRequired": "No",
                        "trackable": {
                            "enabled": "No",
                            "autoBatch": "No"
                        },
                        "visibility": "Parent",
                        "discussionForum": {
                            "enabled": "No"
                        },
                        "mediaType": "content",
                        "languageCode": [
                            "en"
                        ],
                        "version": 2,
                        "versionKey": "1684838561708",
                        "compatibilityLevel": 1,
                        "userConsent": "Yes",
                        "name": "Textbook Unit",
                        "attributions": [],
                        "openForContribution": true,
                        "programId": "5bc03a00-ee67-11ed-a249-d1c2d4936c48",
                        "allowedContentTypes": [
                            "eTextbook",
                            "Explanation Content"
                        ],
                        "origin": "do_11380219749751193617",
                        "originData": {
                            "channel": "01379157488516300825"
                        }
                    }
                }
            },
            "hierarchy": {
                "do_11380283102077747214120": {
                    "name": "book2",
                    "primaryCategory": "Digital Textbook",
                    "children": [
                        "do_11380219749893734419",
                        "do_11380219749751193617"
                    ],
                    "root": true
                },
                "do_11380219749893734419": {
                    "name": "Textbook Unit1",
                    "primaryCategory": "Textbook Unit",
                    "children": [],
                    "root": false
                },
                "do_11380219749751193617": {
                    "name": "Textbook Unit",
                    "primaryCategory": "Textbook Unit",
                    "children": [],
                    "root": false
                }
            }
        }
    }
}'`

@pallakartheekreddy we faced this earlier and seemed data and schema mismatch issue and fixed it. Nothing has changed from that release to current one apart from angular version update (as per my knowledge). It would be great if you can spare some time and help..

**Comment by vinukumar-vs** [comment]
@pallakartheekreddy We are still facing the same issue after verifying the

Schema's loading from blob has have the right metadata with tackable & autobatch init.
Even we have verified all primary category defninations of Digital Textbook & Textbook Unit at Channel level & Global(all) also updated properly...

```
"schema": {
                    "properties": {
                        "trackable": {
                            "type": "object",
                            "properties": {
                                "enabled": {
                                    "type": "string",
                                    "enum": [
                                        "Yes",
                                        "No"
                                    ],
                                    "default": "No"
                                },
                                "autoBatch": {
                                    "type": "string",
                                    "enum": [
                                        "Yes",
                                        "No"
                                    ],
                                    "default": "No"
                                }
                            },
                            "default": {
                                "enabled": "No",
                                "autoBatch": "No"
                            },
                            "additionalProperties": false
                        },
....
```

Any solution what can be the issue here.

[@maheshkumargangula](https://github.com/maheshkumargangula) [@pallakartheekreddy](https://github.com/pallakartheekreddy) @vrayulu

**Reply by vaivk369** [reply]
The newly updated category definition was not loaded due to caching. Re-deploying taxonomy and content-service did the trick.
